valhalla/loki/search.h

```diff
@@ -13,22 +13,27 @@
 namespace valhalla {
 namespace loki {
 
+struct SearchOptions {
+  uint32_t max_reachability_ = 100;
```

Comment:
why do we need this and why do we need to store an extra bool on the candidates we find? each location already has the max_reach assigned to it (minimum_reachabilty). cant we just add a member to that class that says:

```c++
bool high_reach() const {
  return outbound >= min_reach && inbound >= min_reach;
}
```

I fail to see why its needed to keep this information and really make any of the changes in loki search when the candidate is part of the location which already knows what the max reach could be and the candidate knows what the actual reach was.

Reply:
yeah, I also don't like this too much. but, current solutions assumes that we detect big connectivity components. so, if we got location with already set `minimum_reachability` value that is actually low, in this case `high_reach` will return `true` even in cases when this edge lies in a small component.

Reply:
hm, so, probably for route requests we should override `minimum_reachability` with the `max_reachability` value from the config

Reply:
@kevinkreiser `outbound, inbound` values are stored in `PathEdge` class but `minumun_reachability` is kept in `PathLocation`. there are 2 possible solutions I see:
1. define `bool high_reach(uint32_t min_inbound_reach, uint32_t min_outbound_reach) const` method in `PathEdge` class and use `min_outbound_reach_,min_inbound_reach_` values from `Location` class to pass to this method;
2. define `bool high_reach(const PathEdge& path_edge) const` method in `PathLocation` class and use it to detect if an edge is high reachable or not

Reply:
yep either of those would be good, that is what i was thinking too :+1: